docs/source/1.0/spec/aws/aws-cloudformation.rst

```diff
@@ -0,0 +1,672 @@
+-------------------------------------
+
+Summary
+    Indicates that a Smithy resource is a CloudFormation resource.
+Trait selector
+    ``resource``
+Value type
+    ``structure``
+
+The ``aws.cloudformation#resource`` trait is a structure that supports the
+following members:
+
+.. list-table::
+    :header-rows: 1
+    :widths:  10 20 70
+
+    * - Property
+      - Type
+      - Description
+    * - name
+      - ``string``
+      - Provides a custom CloudFormation resource name.
+    * - additionalSchemas
+      - ``list<shapeId>``
+      - A list of additional :ref:`shape IDs <shape-id>` of structures that
+        will have their properties added to the CloudFormation resource.
+
+The following example defines a simple resource that is also a CloudFormation
+resource:
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        namespace smithy.example
+
+        use aws.cloudformation#resource
+
+        service TestService {
+            version: "2020-07-02",
+            resources: [
+                Foo,
+            ],
+        }
+
+        @resource
+        resource Foo {
+            identifiers: {
+                fooId: String,
+            },
+        }
+
+
+The following example provides a value for each of the optional properties:
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        namespace smithy.example
+
+        use aws.cloudformation#resource
+
+        service TestService {
+            version: "2020-07-02",
+            resources: [
+                FooResource,
+            ],
+        }
+
+        @resource(
+            name: "Foo",
+            additionalSchemas: [AdditionalFooProperties])
+        resource FooResource {
+            identifiers: {
+                fooId: String,
+            },
+        }
+
+        structure AdditionalFooProperties {
+            barProperty: String,
+        }
+
+.. _aws.cloudformation#additionalIdentifier-trait:
+
+-------------------------------------------------
+``aws.cloudformation#additionalIdentifier`` trait
+-------------------------------------------------
+
+Summary
+    Indicates that the CloudFormation property generated from this member is an
+    additional identifier for the resource.
+Trait selector
+    ``structure > :test(member > string)``
+
+    *Any structure member that targets a string*
+Value type
+    Annotation trait
+Conflicts with
+    :ref:`aws.cloudformation#writeOnlyProperty-trait`
+
+The following example defines a CloudFormation resource that has the
+``fooAlias`` property as an additional identifier:
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        namespace smithy.example
+
+        use aws.cloudformation#additionalIdentifier
+        use aws.cloudformation#resource
+
+        @resource
+        resource Foo {
+            identifiers: {
+                fooId: String,
+            },
+            get: GetFoo,
+        }
+
+        operation GetFoo {
+            input: GetFooRequest,
+        }
+
+        structure GetFooRequest {
```

Comment:
Are these only found on operations bound to read operations? What are the rules for how this trait actually works?